test-suite-graal/src/test/java/example/GitHubApiClientMock.java

```diff
@@ -0,0 +1,68 @@
+package example;
+
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.starter.client.github.v3.GitHubApiClient;
+import io.micronaut.starter.client.github.v3.GitHubRepository;
+import io.micronaut.starter.client.github.v3.GitHubSecret;
+import io.micronaut.starter.client.github.v3.GitHubSecretsPublicKey;
+import io.micronaut.starter.client.github.v3.GitHubUser;
+import io.micronaut.starter.client.github.v3.GitHubWorkflowRun;
+import io.micronaut.starter.client.github.v3.GitHubWorkflowRuns;
+import jakarta.inject.Singleton;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.transport.URIish;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@Replaces(GitHubApiClient.class)
```

Comment:
We need that since we don't want to create a remote repository in GitHub when a test is executed. Instead of creating the repository in GitHub, the `GitHubApiClientMock` creates the remote repository locally in the `temp` directory. The similar thing has been done in the existing test here: https://github.com/micronaut-projects/micronaut-starter/blob/08cd56dd873285b15eb3b08ca41d6901f557715c/starter-api/src/test/groovy/io/micronaut/starter/api/create/github/GitHubCreateControllerSpec.groovy#L107